src/components/elements/MobileNav.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import { useState } from 'react'
+import { Link } from 'react-router-dom';
+import useDarkMode from 'hooks/dark-mode';
+import Logo from '../../assets/logo.png'
+import lightMode from '../../assets/light-mode.svg'
+import darkMode from '../../assets/dark-mode.svg'
+import closeSymbol from '../../assets/HamX.png'
+import hamMenu from '../../assets/HamReg.png'
+import logoDark from '../../assets/logoDark.png'
+
+
+function MobileNav() {
+  const [menuOpen, setMenuOpen] = useState(false)
+  const [colorTheme, setTheme] = useDarkMode();
+
+
+  return (
+    <>
+      <div className='grid text-right bg-du-gray dark:bg-slate-700 h-16 grid-cols-4 place-content-center place-items-center'>
+        <section className='place-content-center place-items-center'>
+          <Link onClick={() => setMenuOpen(false)} to="/"><img className="object-scale-down object-center h-14 w-16" src={colorTheme === 'dark' ? Logo : logoDark} alt="logo" /></Link>
+        </section>
+        <span className='text-center col-span-2'>
+          <p className='md:text-sm'>Data Science Event Board</p>
+        </span>
+        <button className="col-start-4 pl-9" onClick={() => setMenuOpen((prev) => !prev)}>
+          <img className="inline" src={hamMenu}/>
+        </button>
+      </div>
+      <div className={menuOpen ? 'relative' : 'invisible'}>
+        <div className="dark:bg-du-deepPurple bg-fuchsia-800 z-20 fixed top-0 right-0 pl-8 h-screen text-right w-4/5">
+          <div className='text-white text-2xl grid grid-rows-11 gap-3 place-content-evenly'>
+            <button onClick={() => setMenuOpen((prev) => !prev)} className='pb-12 pt-10 text-right'>
+              Close <img className="inline" src={closeSymbol} />
+            </button>
+            <div>
+              <img
+                src={colorTheme === 'dark' ? darkMode : lightMode}
+                onClick={() => setTheme(colorTheme)}
+                className="float-right pb-2"
+                alt="logo"
+              />
+            </div>
+            <Link onClick={() => setMenuOpen((prev) => !prev)} to="/sign-in">Sign In</Link><br />
```

Comment:
setMenuOpen with arguments gets copied and pasted a lot, create a handler like...

function handleToggleMenu(e) {
  e.preventDefault()
  setMenuOpen(prev => !prev)
}

then...

onClick={handleToggleMenu}

Reply:
Imported useEffect w/ useLocation()